Source/Scene/Camera.js

```diff
@@ -3681,7 +3681,60 @@ var cartoArray = [
   new Cartographic(),
   new Cartographic(),
 ];
-function addToResult(x, y, index, camera, ellipsoid, computedHorizonQuad) {
+
+function rotateQuad(quad, numRotations) {
+  var newQuad = new Array(4);
+  for (var i = 0, j = numRotations; i < 4; i++, j = (j + 1) % 4) {
+    newQuad[j] = quad[i];
+  }
+  return newQuad;
+}
+
+function generalHeading(heading) {
+  if (heading > 45 && heading <= 135) {
+    return "east";
+  }
+  if (heading > 135 && heading <= 225) {
+    return "south";
+  }
+  if (heading > 225 && heading <= 315) {
+    return "west";
+  }
+
+  return "north";
+}
+
+var rotationsFromUp = {
+  up: 0,
+  left: 1,
+  down: 2,
+  right: 3,
+};
+
+var headingQuadrant = {
+  north: 0,
+  east: 1,
+  south: 2,
+  west: 3,
+};
+
+// Takes heading and camera direction (starting from direct view of earth from space)
+// and adjusts the quad array so that its corners roughly match where the camera is pointing.
+//
+// Ex.
+//    - direction is 'left' (looking toward the west with the horizon showing on the upper and lower left corners)
+//    - heading is 80 - camera is rotated clockwise toward the east (still showing horizon on upper and lower left corners).
+//                      At this point, the camera is upside down (up -> south).
+//
+// Given that the corner indeces of the quad are [UL, LL, LR, UR],
+// the quad elements would be 'rotated' like so - [NW, SW, SE, NE] => [SE, NE, NW, SW]
+function changeQuadHeading(heading, direction, quad) {
```

Comment:
This makes sense to me but I wonder if it's a rigorous solution. I might just need some time to think about it some more, but I haven't wrapped my head around adding the number of rotations to the general heading count yet.

@lilleyse what do you think about this math?